### ISSUE TITLE
Usage of pytest plugins can lead to PytestAssertRewriteWarning

### ISSUE THREAD

**Issue by kasium** [reverbc/pylint-pytest#10]
Hi,
I found a quite intresting behaviour when using pytest plugins:

**my_module/fixtures.py**
```python
import pytest
@pytest.fixture
def my_fixture():
    pass
```

**tests/conftest.py**
```python
pytest_plugins= ("my_module.fixtures",)
```

If I now run pylint on my_module with pylint-pytest enabled, I see a _PytestAssertRewriteWarning: Module already imported so cannot be rewritten_. Basically this is not a big issues and the warning can be ignored by using pytest settings.
However, it would be nice, if it would be possible to avoid this issue or if pylint-pytest could ignore this warning by default.

Versions:
- pytest==6.2.2
- pylint==2.7.1
- pylint-pytest==1.0.0

Best regrads,
Kai

**Comment by reverbc** [comment]
Hi Kai, thanks for reporting another issue!

While I'm interested in this issue, I'm having problem to reproduce it.
My local folder structure and command is as below:
![image](https://user-images.githubusercontent.com/783515/110535804-8f918480-80d5-11eb-95ac-166b3dd079ff.png)

I tried a few combinations (moving `my_module` folder around, trigger the `pylint` from different location) but still no luck.

Can you provide the exact folder structure, how and where do you execute the `pylint` command, and the output for me to reproduce and dig deeper?

Thanks

**Comment by kasium** [comment]
Hi,
I'm sorry that a few things were missing. So I created now a sample project which should work: [project.tar.gz](https://github.com/reverbc/pylint-pytest/files/6111464/project.tar.gz).

Here is how to reproduce the issue:
1. Extract the archive and cd into the "rep"
2. virtualenv venv --python=python3.7
3. source venv/bin/activate
4. pip install -e .
5. pylint my_package/ tests/

As far as I can tell, this works for me:

```
(venv) xxx:~/git/rep # pylint my_package/ tests/
************* Module my_package.sub._foo
my_package/sub/_foo.py:2:0: C0304: Final newline missing (missing-final-newline)
my_package/sub/_foo.py:1:0: C0114: Missing module docstring (missing-module-docstring)
my_package/sub/_foo.py:1:0: C0103: Function name "x" doesn't conform to snake_case naming style (invalid-name)
my_package/sub/_foo.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
************* Module my_package.sub.__init__
my_package/sub/__init__.py:1:0: C0304: Final newline missing (missing-final-newline)
************* Module my_package.sub
my_package/sub/__init__.py:1:0: C0114: Missing module docstring (missing-module-docstring)
************* Module my_package.sub.fixtures
my_package/sub/fixtures.py:1:0: C0114: Missing module docstring (missing-module-docstring)
my_package/sub/fixtures.py:4:0: C0116: Missing function or method docstring (missing-function-docstring)
Traceback (most recent call last):
  File "/root/git/rep/venv/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint/__init__.py", line 22, in run_pylint
    PylintRun(sys.argv[1:])
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint/lint/run.py", line 358, in __init__
    linter.check(args)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 863, in check
    self.get_ast, self._iterate_file_descrs(files_or_modules)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 896, in _check_files
    self._check_file(get_ast, check_astroid_module, name, filepath, modname)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 922, in _check_file
    check_astroid_module(ast_node)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1055, in check_astroid_module
    ast_node, walker, rawcheckers, tokencheckers
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1099, in _check_astroid_module
    walker.walk(ast_node)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint/utils/ast_walker.py", line 72, in walk
    callback(astroid)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pylint_pytest/checkers/fixture.py", line 76, in visit_module
    plugins=[fixture_collector],
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 143, in main
    config = _prepareconfig(args, plugins)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 319, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/helpconfig.py", line 100, in pytest_cmdline_parse
    config: Config = outcome.get_result()
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 1003, in pytest_cmdline_parse
    self.parse(args)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 1283, in parse
    self._preparse(args, addopts=addopts)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 1192, in _preparse
    early_config=self, args=args, parser=self._parser
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/root/git/rep/venv/lib/python3.7/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 1070, in pytest_load_initial_conftests
    self.pluginmanager._set_initial_conftests(early_config.known_args_namespace)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 500, in _set_initial_conftests
    self._try_load_conftest(anchor, namespace.importmode)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 508, in _try_load_conftest
    self._getconftestmodules(anchor, importmode)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 536, in _getconftestmodules
    mod = self._importconftest(conftestpath, importmode)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 587, in _importconftest
    self.consider_conftest(mod)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 663, in consider_conftest
    self.register(conftestmodule, name=conftestmodule.__file__)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 443, in register
    self.consider_module(plugin)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 669, in consider_module
    self._import_plugin_specs(getattr(mod, "pytest_plugins", []))
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 676, in _import_plugin_specs
    self.import_plugin(import_spec)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 695, in import_plugin
    self.rewrite_hook.mark_rewrite(importspec)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/assertion/rewrite.py", line 259, in mark_rewrite
    self._warn_already_imported(name)
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/assertion/rewrite.py", line 270, in _warn_already_imported
    stacklevel=5,
  File "/root/git/rep/venv/lib/python3.7/site-packages/_pytest/config/__init__.py", line 1321, in issue_config_time_warning
    warnings.warn(warning, stacklevel=stacklevel)
```

My guess is, that first _my_package.sub-fixtures_ is linted and therefore pylint loads it. Afterwards your plugin calls pytest which then again complains, that _my_package.sub-fixtures_ is already loaded.
Maybe one option could be to run pytest in another process (not sure about that).

Thanks,
Kai

**Comment by reverbc** [comment]
Fixed & released with 1.0.2
